network/topology.go

```diff
@@ -17,5 +18,5 @@ type Topology interface {
 	// One may utilize topology Cache to have a more deterministic endpoint on generating fanout.
 	//
 	// GenerateFanout is not concurrency safe. It is responsibility of caller to lock for it.
-	GenerateFanout(ids flow.IdentityList) (flow.IdentityList, error)
+	GenerateFanout(ids flow.IdentityList, channels ChannelList) (flow.IdentityList, error)
```

Comment:
💡 we input the list of subscribed channels directly to `GenerateFanout`. Hence, we no longer need `SubscriptionManager` as a member field in topology implementations for extracting the list of subscribed channels. This simplifies interactions among networking layer components, and eases up stateful caching.